### PR TITLE
Add functions for obtaining the losses of MLM and MPP.

### PR DESCRIPTION
**Pull request by boru-roylu**

### CODE REVIEW

File: src/run_pretrain.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+
+def model_fn_builder(model_config, init_checkpoint, learning_rate,
+                     num_train_steps, num_warmup_steps, use_tpu,
+                     use_one_hot_embeddings, optimizer, poly_power,
+                     start_warmup_step, learning_rate_schedule):
+  """Returns `model_fn` closure for TPUEstimator."""
+
+  def model_fn(features, labels, mode, params):
+    raise NotImplementedError
```
Comment (wqfcr-google) [inline src/run_pretrain.py:32]: I remember you implemented another model_fn_builder somewhere?

File: src/run_pretrain.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  raise NotImplementedError
+
+
+def get_mpp_output(model_config, input_tensor, output_num_classes, positions,
+                   label_ids, label_weights):
+
+  """Get loss and log probs for the masked patch prediction (MPP)."""
+  input_tensor = tensor_utils.gather_indexes(input_tensor, positions)
+
+  # NOTE (roylu): Do we need layer norm before the linear projection?
```
Comment (wqfcr-google) [inline src/run_pretrain.py:52]: Not sure either. Does VIT include this layer norm before computing the logits?

**Review by wqfcr-google — COMMENTED** [review]
Overall looks good! Just some quick questions.

**Review by jtainslie — APPROVED** [review]

File: src/run_pretrain.py (hunk 2 above)
Comment (boru-roylu) [reply src/run_pretrain.py:52]: I followed the [Pytorch implementation](https://github.com/lucidrains/vit-pytorch/blob/22da26fa4b7d98c2a936b48c1b1dcf83c404dfb8/vit_pytorch/vit.py#L126). Not sure if it's necessary or exists in original implementation.

File: src/run_pretrain.py (hunk 1 above)
Comment (boru-roylu) [reply src/run_pretrain.py:32]: I don't think we have another model_fn_builder.

File: src/run_pretrain.py (hunk 2 above)
Comment (wqfcr-google) [reply src/run_pretrain.py:52]: We could probably try both.

**Review by wqfcr-google — APPROVED** [review]

**Review by santiaontanon-google — APPROVED** [review]
Looks good Roy!
